db/blip_handler.go

```diff
@@ -1049,17 +1049,19 @@ func (bh *blipHandler) processRev(rq *blip.Message, stats *processRevStats) (err
 			injectedAttachmentsForDelta = true
 		}
 
-		deltaSrcMap := map[string]interface{}(deltaSrcBody)
-		err = base.Patch(&deltaSrcMap, newDoc.Body(bh.loggingCtx))
-		// err should only ever be a FleeceDeltaError here - but to be defensive, handle other errors too (e.g. somehow reaching this code in a CE build)
-		if err != nil {
-			// Something went wrong in the diffing library. We want to know about this!
-			base.WarnfCtx(bh.loggingCtx, "Error patching deltaSrc %s with %s for doc %s with delta - err: %v", deltaSrcRevID, revID, base.UD(docID), err)
-			return base.HTTPErrorf(http.StatusUnprocessableEntity, "Error patching deltaSrc with delta: %s", err)
-		}
+		if !revMessage.Deleted() {
```

Comment:
We don't intend to support deltas for deletes (i.e. if `deleted` and `deltaSrc` properties are both specified on an incoming rev message, we should ignore deltaSrc, and perform this check up on line 1016).